libraries/abstractions/common_io/include/iot_sdio.h

```diff
@@ -0,0 +1,714 @@
+{
+    eSdioCardInterruptEvent, /*!< Interrupt received from sdio card */
+    eSdioCardInsertedEvent,  /*!< Sdio card insertion is detected */
+    eSdioCardRemovedEvent,   /*!< Sdio card removal is detected */
+    eSdioCardPowerOnEvent,   /*!< Powering on sdio card */
+    eSdioCardPowerOffEvent,  /*!< Powering off sdio card */
+} IotSdioEventType_t;
+
+/**
+ * @brief IOCTL request types.
+ */
+typedef enum IotSdioIoctlRequest
+{
+    eSDIOSetDriverStrength,   /*!< Set driver strength type
+                               *   input: SDIO card driver strength.
+                               *          input data type IotSdioSdDriverStrength_t */
+    eSDIOSetCardInactive,     /*!< put card in inactive state */
+    eSDIOSwitchToHighSpeed,   /*!< switch to high speed mode if card spports it */
+    eSDIOSetDataBusWidth,     /*!< set sdio card data bus width and change host setting to support it
+                               *   input: sdio bus width, input data type IotSdioBusWidth_t */
+    eSDIOGetCardCapabilities, /*!< request host to learn capabilities of specified card function
+                               *   input: I/O function number, of type IotSdioFunction_t*/
+    eSDIOSetCardDetectParams, /*!< Set card detect parameters
+                               *   input: card detect parameters.
+                               *          input data type IotSdioCardDetectParam_t*/
+    eSDIOSetFuncBlockSize,    /*!< Set transfer block size for a function
+                               *   input: I/O function and block size.
+                               *          input data type IotSdioFuncBlkSize_t*/
+    eSDIOEnableIOFunctionIrq, /*!< enable/disable irq for individual io functions
+                               *   input: I/O function number and enable or disable,
+                               *          input data type IotSdioPerFuncEnable */
+    eSDIOEnableAsyncIrqMode,  /*!< enable/disable asynchronous interrupt mode
+                               *    input: boolean to indicate enable or disable */
+    eSDIOEnableIo,            /*!< enable/disable an I/O function on card
+                               *   input: I/O function number and boolean to indicate enable
+                               *          or disable, input data type IotSdioPerFuncEnable_t */
+    eSDIOSelectIo,            /*!< Select an I/O function for suspend/resume
+                               *   input: I/O function number, input data type IotSdioFunction_t */
+    eSDIOAbortIo,             /*!< Abort an ongoing transfer initiated by the specified I/O function
+                               *   input: I/O function number. Input data type IotSdioFunction_t */
+    eSDIOReadCis,             /*!< Host read list of tuples from card CIS
+                               *   input: I/O function number and tuple list to read,
+                               *          input data type IotSdioReadCis_t*/
+    eSDIOPowerOnOffCard,      /*!< Power on/off card
```

Comment:
Why does the boolean true indicate power on and use the boards default power method? Is there a missing enum?

Reply:
The 2 boolean inputs are separate parameters:
typedef struct IoTSdioPowerOnOffCard
{
    bool bPowerOn;     /*!< True to power on card, false to power off card. */
    bool bUseCallback; /*!< indicate if user callback should be used to power on/off card. */
} IoTSdioPowerOnOffCard_t;